### ISSUE TITLE
Strictness, instanceof and similars

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#229]
`instanceof`, `get_class`, `is_a`, so on, could be dropped in case of strict type. 

**For instance:**

```php
/** @var MyClass $class */

// Warning: MyClass is the strict type of $class.
$class instanceof MyClass;

// Warning: ParentOfMyClass is the strict parent type of $class.
$class instanceof ParentOfMyClass;
```

**Comment by kalessil** [comment]
I can do this for:
- parameters (not overridden) with implicit type hints
- local variables (not overridden) which getting values from a method/function call with implicit return type hint

And relying on DocBlock promise to eat developers time and my support time, so I mentioned implicit declarations.

**Comment by kalessil** [comment]
@rentalhost : if the approach looks good, please schedule this ticket for Ultimate please =)

**Comment by kalessil** [comment]
Goes into "Callable parameter usage violates definition" inspection (Ultimate only)

**Comment by kalessil** [comment]
Done (for parameters only)
